packer/builder/azure/driver_restapi/config.go

```diff
@@ -0,0 +1,200 @@
+
+	// Check values
+	var errs *packer.MultiError
+
+	if c.SubscriptionName == "" {
+		errs = packer.MultiErrorAppend(errs,
+			fmt.Errorf("subscription_name must be specified"))
+	}
+
+	if c.PublishSettingsPath == "" {
+		errs = packer.MultiErrorAppend(errs, fmt.Errorf("publish_settings_path must be specified"))
+	}
+
+	if c.StorageAccount == "" {
+		errs = packer.MultiErrorAppend(errs, fmt.Errorf("storage_account must be specified"))
+	}
+
+	if _, err := os.Stat(c.PublishSettingsPath); err != nil {
+		errs = packer.MultiErrorAppend(errs, fmt.Errorf("publish_settings_path is not a valid path: %s", err))
+	}
+
+	osTypeIsValid := false
+	osTypeArr := []string{
+		targets.Linux,
+		targets.Windows,
+	}
+
+	for _, osType := range osTypeArr {
+		if c.OsType == osType {
+			osTypeIsValid = true
+			break
+		}
+	}
+
+	if !osTypeIsValid {
+		errs = packer.MultiErrorAppend(errs,
+			fmt.Errorf("os_type is not valid, must be one of: %v", osTypeArr))
+	}
+
+	if c.OsImageLabel == "" {
+		errs = packer.MultiErrorAppend(errs, fmt.Errorf("os_image_label must be specified"))
+	}
+
+	if c.Location == "" {
+		errs = packer.MultiErrorAppend(errs, fmt.Errorf("location must be specified"))
+	}
+
+	sizeIsValid := false
+	instanceSizeArr := []string{
+		targets.ExtraSmall,
+		targets.Small,
+		targets.Medium,
+		targets.Large,
+		targets.ExtraLarge,
+		targets.A5,
+		targets.A6,
+		targets.A7,
+		targets.A8,
+		targets.A9,
+		targets.Standard_D1,
+		targets.Standard_D2,
+		targets.Standard_D3,
+		targets.Standard_D4,
+		targets.Standard_D11,
+		targets.Standard_D12,
+		targets.Standard_D13,
+		targets.Standard_D14,
+		targets.Standard_DS1,
+		targets.Standard_DS2,
+		targets.Standard_DS3,
+		targets.Standard_DS4,
+		targets.Standard_DS11,
+		targets.Standard_DS12,
+		targets.Standard_DS13,
+		targets.Standard_DS14,
+		targets.Standard_G1,
+		targets.Standard_G2,
+		targets.Standard_G3,
+		targets.Standard_G4,
+		targets.Standard_G5,
+	}
+
+	for _, instanceSize := range instanceSizeArr {
+		if c.InstanceSize == instanceSize {
+			sizeIsValid = true
+			break
+		}
+	}
+
+	if !sizeIsValid {
+		errs = packer.MultiErrorAppend(errs, fmt.Errorf("instance_size is not valid, must be one of: %v", instanceSizeArr))
+	}
+
+	if c.UserImageLabel == "" {
+		errs = packer.MultiErrorAppend(errs, fmt.Errorf("user_image_label must be specified"))
+	}
+
+	// check
+	pattern := "^[A-Za-z][A-Za-z0-9-_.]*[A-Za-z0-9]$"
+	value := c.UserImageLabel
+
+	match, _ := regexp.MatchString(pattern, value)
+	if !match {
+		errs = packer.MultiErrorAppend(errs, fmt.Errorf("user_image_label is not valid, it should follow the pattern %s", pattern))
+	}
+
+	c.userImageName = utils.DecorateImageName(c.UserImageLabel)
+	c.tmpContainerName = utils.BuildContainerName()
+
+	// for Win  - the computer name cannot be more than 15 characters long
+	const tmpServiceNamePrefix = "PkrSrv"
+	const tmpVmNamePrefix = "PkrVM"
+	randSuffix := utils.BuildAzureVmNameRandomSuffix(tmpVmNamePrefix)
+
+	if c.tmpVmName == "" {
```

Comment:
Since c isn't returned if err has a value, should we check err and not bother updating tmp* values in c in the next few lines?. Might be alright if this is needed for debugging/logging.